### PR TITLE
Tracks: Don't track if Jetpack is not present or in dev env.

### PR DESCRIPTION
**Pull request by timmyc**
One small follow-up to #452 based on some conversations and feedback on prior PRs. This branch adds Jetpack being installed/activated as a requirement to the logic to determine if tracking is enabled ( per @warrendholmes ), it also adds a check to determine if the node process is `development` and if so, not to record tracks so we don't gather data from our development usage.

__To Test__
- Open up an analytics page and verify no `t.gif` settings are recorded
- Comment out the changes in `client/lib/tracks.js`, or start up your node env in non-development mode
- Verify that `t.gif` is being recorded on analytics pages
- Disable Jetpack
- Verify that `t.gif` is not being recorded.

### CODE REVIEW

File: lib/client-assets.php [hunk 1]
```diff
@@ -55,9 +55,10 @@ function wc_admin_register_script() {
 	$content     = 'wp.i18n.setLocaleData( ' . json_encode( $locale_data ) . ', "wc-admin" );';
 	wp_add_inline_script( 'wc-components', $content, 'before' );
 
-	// Add Tracks script to the DOM if tracking is opted in.
+	// Add Tracks script to the DOM if tracking is opted in, and Jetpack is installed/activated.
 	$tracking_enabled = 'yes' === get_option( 'woocommerce_allow_tracking', 'no' );
-	if ( $tracking_enabled ) {
+	$jetpack_active = is_plugin_active( 'jetpack/jetpack.php' );
```
Comment (ryelle) [inline lib/client-assets.php]: Maybe check for `JETPACK__VERSION` in case the plugin is installed in a different folder?

**Comment by timmyc** [comment]
Thanks for the tip @ryelle, updated to use `JETPACK__VERSION`

**Comment by allendav** [comment]
Would be nice to reference changes that are probably needed to the Jetpack privacy policy. Is there an issue or PR on the Jetpack repo that can be linked from here?

File: lib/client-assets.php [hunk 2]
```diff
@@ -55,9 +55,9 @@ function wc_admin_register_script() {
 	$content     = 'wp.i18n.setLocaleData( ' . json_encode( $locale_data ) . ', "wc-admin" );';
 	wp_add_inline_script( 'wc-components', $content, 'before' );
 
-	// Add Tracks script to the DOM if tracking is opted in.
+	// Add Tracks script to the DOM if tracking is opted in, and Jetpack is installed/activated.
 	$tracking_enabled = 'yes' === get_option( 'woocommerce_allow_tracking', 'no' );
-	if ( $tracking_enabled ) {
+	if ( $tracking_enabled && defined( 'JETPACK__VERSION' ) ) {
```
Comment (allendav) [inline lib/client-assets.php]: Could we also NOT embed tracking if Jetpack is being used in development mode? https://jetpack.com/support/development-mode/

**Comment by timmyc** [comment]
@allendav the convo about changes needed to TOS are being discussed in p2y3YZ-3s3-p2

**Comment by timmyc** [comment]
Based off our convo in the hangout this morning, going to close this one out.

**Pull request closed without merging**